### PR TITLE
add kbd75rgb support

### PR DESCRIPTION
**Pull request by moyi4681**
add kbd75rgb support

### CODE REVIEW

File: keyboards/kbdfans/kbd75rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+#define ENABLE_RGB_MATRIX_PIXEL_FLOW
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/config.h:77]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspendedz
```

File: keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd75rgb/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/kbdfans/kbd75rgb/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# KBD75RGB
+
+A customizable 75% RGB keyboard.
+
+* Keyboard Maintainer: [moyi4681](https://github.com/moyi4681)
+* Hardware Supported: KBDFANS
+* Hardware Availability: [kbdfans](https://kbdfans.myshopify.com/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/kbdfans/kbd75rgb/rules.mk [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = lufa-ms
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes        # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/rules.mk:9]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kbdfans/kbd75rgb/rules.mk [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+MIDI_ENABLE = no               # MIDI support
+BLUETOOTH_ENABLE = no          # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no              # Audio output on port C6
+FAUXCLICKY_ENABLE = no         # Use buzzer to emulate clicky switches
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+LAYOUTS = 75_ansi
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/rules.mk:28]:
```suggestion
FORCE_NKRO = yes
NKRO_ENABLE = yes              # USB Nkey Rollover
BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
AUDIO_ENABLE = no              # Audio output on port C6
RGB_MATRIX_ENABLE = yes        # Use RGB matrix
RGB_MATRIX_DRIVER = WS2812
LTO_ENABLE = yes
LAYOUTS = 75_ansi

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/kbd75rgb/config.h [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+#define USB_SUSPEND_WAKEUP_DELAY 5000
+#define VIA_EEPROM_LAYOUT_OPTIONS_SIZE 2
+
+#define RGB_DI_PIN C7
+#define RGBLED_NUM 84
+#define DRIVER_LED_TOTAL RGBLED_NUM
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/config.h:48]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kbdfans/kbd75rgb/config.h [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+#define ENABLE_RGB_MATRIX_RAINDROPS
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+#define ENABLE_RGB_MATRIX_PIXEL_FLOW
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
+#ifdef RGB_MATRIX_ENABLE
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/config.h:76]:
```suggestion
```

File: keyboards/kbdfans/kbd75rgb/config.h [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspendedz
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 150 // limits maximum brightness of LEDs to 150 out of 255. Higher may cause the controller to crash.
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+#define RGB_MATRIX_HUE_STEP 8
+#define RGB_MATRIX_SAT_STEP 8
+#define RGB_MATRIX_VAL_STEP 8
+#define RGB_MATRIX_SPD_STEP 10
+#endif
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/config.h:85]:
```suggestion
```

File: keyboards/kbdfans/kbd75rgb/rules.mk [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+BOOTLOADER = lufa-ms
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no          # Mouse keys
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+FORCE_NKRO = yes
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/rules.mk:14]:
```suggestion
```

File: keyboards/kbdfans/kbd75rgb/rules.mk [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+AUDIO_ENABLE = no              # Audio output on port C6
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+LAYOUTS = 75_ansi
```
Comment (drashna) [inline keyboards/kbdfans/kbd75rgb/rules.mk:22]:
```suggestion

LAYOUTS = 75_ansi
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c:21]:
```suggestion
  [0] = LAYOUT_75_ansi(
```

File: keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_PSCR,  KC_INS, KC_HOME,
+    KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_BSPC,  KC_END,
+    KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+    KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,            KC_ENT,   KC_PGDN,
+    KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,            KC_UP,    KC_DEL,
+    KC_LCTL,  KC_LGUI,  KC_LALT,                               KC_SPC,              KC_RALT,            KC_RGUI,  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT
+  ),
+
+  [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c:30]:
```suggestion
  [1] = LAYOUT_75_ansi(
```

File: keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  RESET,    _______,   
+    _______,  RGB_TOG,  RGB_MOD,  RGB_HUI,  RGB_HUD,  RGB_SAI,  RGB_SAD,  RGB_VAI,  RGB_VAD,  _______,  _______,  _______,  _______,  _______,  _______,          
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,  _______,  BL_DEC,   BL_TOGG,  BL_INC,   BL_STEP,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,                                _______,            _______,            _______,  _______,  _______,  _______,  _______
+  ),
+  
+   [2] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c:39]:
```suggestion
   [2] = LAYOUT_75_ansi(
```

File: keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  RESET,    _______,   
+    _______,  RGB_TOG,  RGB_MOD,  RGB_HUI,  RGB_HUD,  RGB_SAI,  RGB_SAD,  RGB_VAI,  RGB_VAD,  _______,  _______,  _______,  _______,  _______,  _______,          
+    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,  _______,  BL_DEC,   BL_TOGG,  BL_INC,   BL_STEP,  _______,  _______,  _______,  _______,            _______,  _______,
+    _______,  _______,  _______,                                _______,            _______,            _______,  _______,  _______,  _______,  _______
+  ),
+   [3] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75rgb/keymaps/default/keymap.c:47]:
```suggestion
   [3] = LAYOUT_75_ansi(
```

File: keyboards/kbdfans/kbd75rgb/rules.mk [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no          # Mouse keys
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+AUDIO_ENABLE = no              # Audio output on port C6
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
```
Comment (noroadsleft) [inline keyboards/kbdfans/kbd75rgb/rules.mk:18]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
RGB_MATRIX_ENABLE = yes     # Use RGB matrix
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**